.buildkite/premerge.steps.yaml

```diff
@@ -38,6 +50,11 @@ steps:
     artifact_paths:
       - "UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
 
+  - label: "enforce version restrictions"
+    command: "ci/check-version-file.sh"
+    <<: *script_runner
```

Comment:
`check-version-file` script is fast to run, so I think it would be good to run it before the other steps and fail the build without running anything else if it fails.

Reply:
Similar to above - We want this to be a separate step so that this can fail on its own, while still getting information about whether the GDK build in general succeeds.
As for the order, buildkite steps actually run in parallel unless you put in a -wait step, so the order only matters for how this is displayed. Since building is the main tasks, I've put them higher up to be more visible.

Reply:
Ok, that makes sense. I was thinking that by the point when you have a PR against `release` or `preview` you will probably have pinned an Engine version and would not be iterating as much anymore. But I guess makes sense to give people as much information about failures/successes as possible.